README.md

```diff
@@ -1 +1,119 @@
 # egghead-instructor-center
+
+_An app for egghead instructors to do instructor things_
+
+[ ![Codeship Status for eggheadio/egghead-instructor-center](https://app.codeship.com/projects/3a4afe00-8808-0134-e6cc-2e5dfce30ebc/status?branch=master)](https://app.codeship.com/projects/183842)
+
+---
+
+# Getting started
+
+- Ensure you have Git, Node, and Yarn installed
+- Clone this repo and `cd` into it
+- `yarn` to install packages
+- `yarn test` to run tests
+- `yarn dev` to run app
+- Open [localhost:3000](http://localhost:3000) to view app
+
+---
+
+# Conventions
+
+## Package management
+
+Yarn is used for package management. Use the [yarn cli](https://yarnpkg.com/en/docs/usage) to add/remove/update packages which updates both the `package.json` and `yarn.lock` to ensure consistent package installs. 
+
+## Scripts
+
+Yarn is used for running scripts. Use `yarn {script}` to run them. The core of the scripts extend `react-scripts` (from Create React App) so that this project doesn't have to manage compiling, linting, or testing; it generally gets all [Create React App updates](https://github.com/facebookincubator/create-react-app/releases) for free by running `yarn upgrade react-scripts@latest`. Avoid ejecting the project for this reason if at all possible.
+
+## Components
+
+Each directory inside `src` is a **component**. A component is a directory organized _by feature_. They look like this:
+
+```
+SomeComponentName/
+    index.js (entry point)
+    components/ (optional sub-components)
+    utils/ (optional utility modules)
+```
+
+## Screens
+
+Some components are also **screens**. A screen is just a component that additionally corresponds with a route and is generally connected to endpoints. They are also known as "Smart", "Container", or "Connected" components. They look like this:
+
+```
+SomeScreenName/
+    index.js (entry point)
+    screens/ (optional sub-screens paired with sub-routes)
+    components/ (optional sub-components)
+    utils/ (optional utility modules)
+```
+
+## Routing
+
+[React Router](https://reacttraining.com/react-router/) is used for routing. URLs follow the Rails standard routing scheme, which uses plural entities named with a noun. For example:
+
+```
+/lessons
+/lessons/new
+/lessons/{id}
+/lessons/{id}/edit
+```
+
+Routes correspond to screen components. For example, `src/screens/Lessons` is used for `/lessons` and `src/screens/Lessons/screens/new` is used for `/lessons/new`.
+
+## Promotion
+
+All resources are eligible for **promotion** to facilitate code reuse. If a resource is shared by multiple files the principle of _least common ancestor_ applies and that shared resource is _promoted_ to the least common ancestor’s directory. A resource can be promoted further - outside of the project - into [`egghead-ui`](https://styleguide.egghead.io) or npm to be used across projects.
```

Comment:
The general idea is that when you sit down to write a new feature, it will consist of components. We aren't going to put a lot of thought into reuse when we are writing initially, but often you will want a component in a sibling screen/component. It is then **promoted** to a common ancestor. This continues up to the top level of the project, but you can promote our of the project too. In our case, that would mean `egghead-ui` which is built as a reusable component library shared *across multiple projects*.

Building reusable stuff is hard, and the more robust and reusable it is, the more complicated it is to develop. So we start with one-offs and **promote** as required/needed. This lets us incrementally evolve reusable components instead of participating in "big design up front" for something that we may never really need to reuse.

Does that clarify at all?